x/gamm/pool-models/balancer/amm.go

```diff
@@ -61,7 +63,28 @@ func (p Pool) CalcOutAmtGivenIn(
 	return sdk.NewDecCoinFromDec(tokenOutDenom, tokenAmountOut), nil
 }
 
-// calcInAmtGivenOut calculates token to be provided, fee added,
+// SwapOutAmtGivenIn is a mutative method for CalcOutAmtGivenIn, which includes the actual swap
+func (p Pool) SwapOutAmtGivenIn(
```

Comment:
p Needs to be mutative
```suggestion
func (p *Pool) SwapOutAmtGivenIn(
```